tyssue/core/history.py

```diff
@@ -151,14 +159,120 @@ def retrieve(self, time):
             f"{self.sheet.identifier}_{time:04.3f}", sheet_datasets, self.sheet.specs
         )
 
+    def closefile(self):
+        pass
+
     def __iter__(self):
 
         for t in self.time_stamps:
             sheet = self.retrieve(t)
             yield t, sheet
 
 
+class HistoryHdf5(History):
+    """ This class handles recording and retrieving time series
+    of sheet objects.
+
+    """
+
+    def __init__(self, sheet, extra_cols=None, path="", overwrite=False):
+        """Creates a `SheetHistory` instance.
+
+        Parameters
+        ----------
+        sheet : a :class:`Sheet` object which we want to record
+        extra_cols : dictionnary with sheet.datasets as keys and list of
+            columns as values. Default None
+        """
+        if path is None:
+            warnings.warn(
+                "No directory is giving. The HDF5 file will be saving in the working directory.")
+            self.path = os.os.getcwd()
+        else:
+            self.path = path
+
+        if os.path.exists(os.path.join(self.path, 'out.hf5')):
+            if overwrite:
+                self.hf5file = pd.HDFStore(
+                    os.path.join(self.path, 'out.hf5'), 'w')
+                warnings.warn("The file already exist and will be overwrite.")
+            else:
+                expend = 1
+                while True:
+                    expend += 1
+                    new_file_name = "out{}.hf5".format(str(expend))
+                    if os.path.exists(os.path.join(self.path, new_file_name)):
+                        continue
+                    else:
+                        self.hf5file = pd.HDFStore(
+                            os.path.join(self.path, new_file_name), 'w')
+                        warnings.warn(
+                            "The file already exist and the new filename is {}".format(new_file_name))
+                        break
+
+        else:
+            self.hf5file = pd.HDFStore(os.path.join(self.path, 'out.hf5'), 'w')
+        History.__init__(self, sheet, extra_cols)
+
+    def record(self, to_record=None, time_stamp=None):
+        """Appends a copy of the sheet datasets to the history instance.
+
+        Parameters
+        ----------
+        to_report : list of strings, default ['vert']
+            the datasets from self.sheet to be saved
+
+        """
+        if to_record is None:
+            to_record = ["vert"]
+
+        if time_stamp is not None:
+            self.time = time_stamp
+        else:
+            self.time += 1
+
+        for element in to_record:
+            df = getattr(self.sheet, "{}_df".format(element))
+            times = pd.Series(
+                np.ones((df.shape[0],)) * self.time, name="time")
+            df = pd.concat([df, times], ignore_index=False,
```

Comment:
aren't we falling back to having a big in memory object?

Reply:
Yes, I execute a simulation with 30min duration with dt as 0.1, and the final file made 65Go....
Maybe we can specify specificaly the number of time we want to record ?

Reply:
The fact that the file is big is not a problem if you don't saturate the RAM and can run aribtrary long simulation. I thought (wrongly) the `concat` here added the current data to the history dataframe, that would grow in the RAM... Nevertheless, I agree it would be nice to have a way to only write a subset of the history. Maybe a `record_every` or `record_frequency` argument somewhere.

Reply:
Do we add record arg also in the History class ? Or only in HistoryHdf5 class?

Reply:
In the base class